magicctapipe/scripts/lst1_magic/lst1_magic_create_irf.py

```diff
@@ -323,230 +84,371 @@ def create_irf(
+    eng_bins_stop = u.Quantity(config_irf["energy_bins"]["stop"])
+    eng_bins_n_edges = config_irf["energy_bins"]["n_edges"]
 
-    energy_bins = np.geomspace(**config_irf["energy_bins"]) * u.TeV
+    logger.info(
+        "\nEnergy bins (log space):"
+        f"\n\tstart: {eng_bins_start}"
+        f"\n\tstop: {eng_bins_stop}"
+        f"\n\tn_edges: {eng_bins_n_edges}"
+    )
+
+    energy_bins = u.Quantity(
+        value=np.geomspace(
+            start=eng_bins_start.to_value(u.TeV).round(3),
+            stop=eng_bins_stop.to_value(u.TeV).round(3),
+            num=eng_bins_n_edges,
+        ),
+        unit=u.TeV,
+    )
 
-    logger.info("\nMigration bins (log space):")
-    for key, value in config_irf["migration_bins"].items():
-        logger.info(f"\t{key}: {value}")
+    migra_bins_start = config_irf["migration_bins"]["start"]
+    migra_bins_stop = config_irf["migration_bins"]["stop"]
+    migra_bins_n_edges = config_irf["migration_bins"]["n_edges"]
 
-    migration_bins = np.geomspace(**config_irf["migration_bins"])
+    logger.info(
+        "\nMigration bins (log space):"
+        f"\n\tstart: {migra_bins_start}"
+        f"\n\tstop: {migra_bins_stop}"
+        f"\n\tn_edges: {migra_bins_n_edges}"
+    )
 
-    logger.info("\nFoV offset bins (linear space, unit = deg):")
-    for key, value in config_irf["fov_offset_bins"].items():
-        logger.info(f"\t{key}: {value}")
+    migration_bins = np.geomspace(migra_bins_start, migra_bins_stop, migra_bins_n_edges)
 
-    fov_offset_bins = np.linspace(**config_irf["fov_offset_bins"]) * u.deg
+    if is_bkg_mc:
 
-    if not only_gamma_mc:
-        logger.info("\nBackground FoV offset bins (linear space, unit = deg):")
-        for key, value in config_irf["bkg_fov_offset_bins"].items():
-            logger.info(f"\t{key}: {value}")
+        bkg_fov_bins_start = u.Quantity(config_irf["bkg_fov_offset_bins"]["start"])
+        bkg_fov_bins_stop = u.Quantity(config_irf["bkg_fov_offset_bins"]["stop"])
+        bkg_fov_bins_n_edges = config_irf["bkg_fov_offset_bins"]["n_edges"]
 
-        bkg_fov_offset_bins = np.linspace(**config_irf["bkg_fov_offset_bins"]) * u.deg
+        logger.info(
+            "\nBackground FoV offset bins (linear space):"
+            f"\n\tstart: {bkg_fov_bins_start}"
+            f"\n\tstop: {bkg_fov_bins_stop}"
+            f"\n\tn_edges: {bkg_fov_bins_n_edges}"
+        )
+
+        bkg_fov_offset_bins = u.Quantity(
+            value=np.linspace(
+                start=bkg_fov_bins_start.to_value(u.deg).round(1),
+                stop=bkg_fov_bins_stop.to_value(u.deg).round(1),
+                num=bkg_fov_bins_n_edges,
+            ),
+            unit=u.deg,
+        )
 
     extra_header = {
         "TELESCOP": "CTA-N",
         "INSTRUME": "LST-1_MAGIC",
         "FOVALIGN": "RADEC",
-        "PNT_ZD": (pnt_gamma[0], "deg"),
-        "PNT_AZ": (pnt_gamma[1], "deg"),
+        "PNT_ZD": (pnt_gamma[0].value, "deg"),
```

Comment:
maybe safer: 
```
.to_value('deg')
```

Reply:
I used `.value` because I'm sure that it is already in degree (defined in `load_mc_dl2_data_file`), but OK, I agree that `.to_value` would be safer, so I changed it.